particle_counter/main/hmi.m

```diff
@@ -352,22 +355,28 @@ function pushbutton1_Callback(hObject, eventdata, handles)
     if ~exist(['dataset_evaluation\', TodayDate],'dir')
         mkdir('dataset_evaluation\', TodayDate)
     end
+     strSystemTime = m2xdate(now);
+%      contamination_hist(:, Flag.Time) = [strSystemTime, calculation.contam_level_gram_per_liter, calculation.contam_level_num_per_sample];
+    histogram = []; 
+    for k = 1:20
+         histogram = [histogram sum(calculation.equiv_diam<k/10 & calculation.equiv_diam>(k-1)/10)];
```

Comment:
My point is at the current revision the histogram results would be always based on the `equiv_diam`, right?

The following link may be helpful: 

http://www.mathworks.com/help/matlab/ref/setfield.html